x-pack/plugins/security_solution/server/lib/telemetry/endpoint_metadata_telemetry.ts

```diff
@@ -0,0 +1,124 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { ISavedObjectsRepository } from 'src/core/server';
+import { AgentMetadata } from './../../../../ingest_manager/common/types/models/agent';
+import { AGENT_SAVED_OBJECT_TYPE } from './../../../../ingest_manager/common/constants/agent';
+import { AgentSOAttributes } from '../../../../ingest_manager/common';
+
+export interface EndpointMetadataOSTelemtry {
+  name: string;
+  version: string;
+  count: number;
+}
+
+export interface EndpointPoliciesTelemetry {
+  malware: {
+    success: number;
+    warning: number;
+    failure: number;
+  };
+}
+
+export interface IngestManagerLocalMetadata extends AgentMetadata {
+  host: {
+    id: string;
+  };
+  os: {
+    name: string;
+    version: string;
+    full: string;
+  };
+}
+
+export interface EndpointMetadataTelemetry {
+  endpoints: {
+    installed: number;
+    last_24_hours: number;
+  };
+  os: Record<string, EndpointMetadataOSTelemtry>;
+  policies: EndpointPoliciesTelemetry;
+}
+
+export const getDefaultEndpointMetadataTelemetry = (): EndpointMetadataTelemetry => ({
+  endpoints: { installed: 0, last_24_hours: 0 },
+  os: {} as Record<string, EndpointMetadataOSTelemtry>,
+  policies: {
```

Comment:
Should we remove these if we're not planning on implementing them for 7.9?

Reply:
Yea, I'll take them out unless something magical happens in the next day or two 😂